src/fetching/getEvent.ts

```diff
@@ -1,16 +1,52 @@
 import { defaultFetch } from '.';
-import { ZetkinEvent } from '../types/zetkin';
+import { ZetkinEvent, ZetkinEventResponse } from '../types/zetkin';
 
 export default function getEvent(orgId : string, eventId : string, fetch = defaultFetch) {
     return async () : Promise<ZetkinEvent> => {
         const cRes = await fetch(`/orgs/${orgId}/campaigns`);
         const cData = await cRes.json();
 
+        const oRes = await fetch(`/orgs/${orgId}`);
+        const oData = await oRes.json();
+
+        const bookedRes = await fetch('/users/me/actions');
+        const bookedData = await bookedRes.json();
+
+        const rRes = await fetch('/users/me/action_responses');
+        const rData = await rRes.json();
+
+        const org = {
+            id: oData.data.id,
+            title: oData.data.title,
+        };
+
         for (const obj of cData.data) {
             const eventsRes = await fetch(`/orgs/${orgId}/campaigns/${obj.id}/actions`);
             const campaignEvents = await eventsRes.json();
-            const eventData = campaignEvents.data.find((event : ZetkinEvent) => event.id.toString() === eventId);
-            if (eventData) {
+
+            const campaignEvent = campaignEvents.data?.find((event : ZetkinEvent) =>
+                event.id.toString() === eventId);
+
+            if (campaignEvent) {
+                let isBookedEvent = false;
+                let hasEventResponse = false;
+
+                if (bookedData.data) {
+                    isBookedEvent = bookedData.data.some((booked : ZetkinEvent) =>
+                        booked.id === campaignEvent.id);
+                }
```

Comment:
These if statements should be unnecessary. You could use `bookedData.data?.some()` instead.

Reply:
Didn't work when I tried it but will give it another try!